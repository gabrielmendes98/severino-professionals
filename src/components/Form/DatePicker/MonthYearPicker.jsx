import { DatePicker } from '@material-ui/pickers';
import PropTypes from 'prop-types';
import moment from 'moment';
import { useField, useFormikContext } from 'formik';
import { handleValue } from './util';

const MonthYearPicker = ({ name, id, ...otherProps }) => {
  const { setFieldValue } = useFormikContext();
  const [field, meta] = useField(name);

  const onChange = value => setFieldValue(name, moment(value).format());

  const configTextfield = {
    ...field,
    autoOk: true,
    fullWidth: true,
    id: id || name,
    onChange,
    views: ['year', 'month'],
    inputVariant: 'outlined',
    size: 'small',
    value: handleValue(field.value),
    ...otherProps,
  };

  if (meta && meta.touched && meta.error) {
    configTextfield.error = true;
    configTextfield.helperText = meta.error;
  }

  return (
    <DatePicker
      {...configTextfield}
      DialogProps={{ id: `${id || name}Dialog` }}
    />
  );
};

MonthYearPicker.propTypes = {
  id: PropTypes.string,
  name: PropTypes.string.isRequired,
};

export default MonthYearPicker;
