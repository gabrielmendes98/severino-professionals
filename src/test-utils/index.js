/* eslint-disable react/prop-types */
import { render, configure, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Form, Formik } from 'formik';
import MainProvider from 'commons/providers/MainProvider';

configure({ testIdAttribute: 'id' });

const Wrapper = ({ children }) => (
  <>
    <MainProvider>{children}</MainProvider>
    <div id="modal"></div>
  </>
);
const RouteWrapper = ({ children }) => (
  <BrowserRouter>
    <Wrapper>{children}</Wrapper>
  </BrowserRouter>
);
const FormikWrapper = ({ children, initialValues, ...config }) => (
  <Formik initialValues={initialValues} {...config}>
    <Form>{children}</Form>
  </Formik>
);

const customRender = (ui, options) =>
  render(ui, { wrapper: Wrapper, ...options });

const renderWithRouter = (ui, { route = '/', ...options } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return render(ui, { wrapper: RouteWrapper, ...options });
};

const asFragment = ui => {
  const { asFragment } = customRender(ui);
  return asFragment(ui);
};

const toMatchSnapshot = (Component, { useRouter = false } = {}) => {
  const wrapper = asFragment(
    useRouter ? <BrowserRouter>{Component}</BrowserRouter> : Component,
  );
  expect(wrapper).toMatchSnapshot();
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export {
  customRender as render,
  screen,
  toMatchSnapshot,
  asFragment,
  renderWithRouter,
  FormikWrapper,
};
