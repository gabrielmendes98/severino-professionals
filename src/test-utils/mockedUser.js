const mockedUser = {
  id: '9c6cb22e-f284-4ca4-9a88-d64ece7b896c',
  createdAt: '2021-09-02T03:55:52.338Z',
  updatedAt: '2021-09-02T03:55:52.338Z',
  email: 'test@test.com',
  name: 'Testerson da Silva',
  description: null,
  avatarUrl: 'www.avatarurlfake.com',
  phone: '(11) 11111-1112',
  state: 'Minas Gerais',
  city: '√Åguas Formosas',
  hasWhatsapp: false,
};
export default mockedUser;
