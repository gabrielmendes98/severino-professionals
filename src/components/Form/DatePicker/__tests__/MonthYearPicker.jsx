import { toMatchSnapshot, FormikWrapper } from 'test-utils';
import MonthYearPicker from '../MonthYearPicker';

it('should match snapshot', () => {
  toMatchSnapshot(
    <FormikWrapper initialValues={{ startDate: '' }}>
      <MonthYearPicker name="startDate" label="Data de início" />
    </FormikWrapper>,
  );
});

it('should match snapshot with error', () => {
  toMatchSnapshot(
    <FormikWrapper
      initialValues={{ startDate: '' }}
      initialErrors={{ startDate: 'Error test' }}
      initialTouched={{ startDate: true }}
    >
      <MonthYearPicker name="startDate" label="Data de início" />
    </FormikWrapper>,
  );
});
