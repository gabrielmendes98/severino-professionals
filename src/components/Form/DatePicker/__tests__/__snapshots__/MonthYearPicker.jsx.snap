// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      format="MMMM YYYY"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
        data-shrink="false"
        for="startDate"
        id="startDate-label"
      >
        Data de início
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-invalid="false"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="startDate"
          name="startDate"
          readonly=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-3"
          >
            <span>
              Data de início
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
  </form>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;

exports[`should match snapshot with error 1`] = `
<DocumentFragment>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      format="MMMM YYYY"
    >
      <label
        class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined Mui-error Mui-error"
        data-shrink="false"
        for="startDate"
        id="startDate-label"
      >
        Data de início
      </label>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-marginDense MuiOutlinedInput-marginDense"
      >
        <input
          aria-describedby="startDate-helper-text"
          aria-invalid="true"
          class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
          id="startDate"
          name="startDate"
          readonly=""
          type="text"
          value=""
        />
        <fieldset
          aria-hidden="true"
          class="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Data de início
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        class="MuiFormHelperText-root MuiFormHelperText-contained Mui-error MuiFormHelperText-marginDense"
        id="startDate-helper-text"
      >
        Error test
      </p>
    </div>
  </form>
  <div
    class="Toastify"
  />
</DocumentFragment>
`;
